packages/react/src/lib/factories.ts

```diff
@@ -230,9 +227,16 @@ function createShorthandFromValue({
   const { generateKey = true } = options
 
   // Use key or generate key
-  if (generateKey && _.isNil(props.key) && valIsPrimitive) {
-    // use string/number shorthand values as the key
-    props.key = value
+  if (generateKey && _.isNil(props.key)) {
+    if (valIsPrimitive) {
+      // use string/number shorthand values as the key
+      props.key = value
+    }
+
+    if (valIsReactElement) {
+      // use the key from React Element
+      props.key = (value as React.ReactElement).key
+    }
```

Comment:
I am not sure about this place, but usage like below looks understandable to me:

```jsx
<List
  items={[
    'string', // key will be 'string'
    10,  // key will be '10'
    <span key='foo' />, // key will be 'foo'
    { key: 'foo', content: <span /> }, // otherwise only this option will be available 👎 
  ]}
/>
```

Reply:
We decided to keep behavior with `key` definition